src/endpoint.cpp

```diff
@@ -678,7 +678,7 @@ int UartEndpoint::read_msg(struct buffer *pbuf, int *target_sysid, int *target_c
 
     if (_change_baud_timeout != nullptr && ret == ReadOk) {
         log_info("Baudrate %lu responded, keeping it", _baudrates[_current_baud_idx]);
-        Mainloop::get_instance().del_timeout(_change_baud_timeout);
```

Comment:
seems wrong, nops

Reply:
Because del_timeout doesn't change any member variable of the class, the method should be static. So calling it via `Classname::static_method()` is actually the right way.

Reply:
But that is more an implementation detail than anything else. It's a pain if the user of an API has to guess on each method if it's a class or object method. If all of them were bound to the class, or just some obvious ones like factory-like methods, then it'd be fine. If the compiler want's to optimize it away, fine, but don't enforce it on the source.

Reply:
OK, I reverted and disabled the whole clang-tidy rule for now.